idlparser/src/ffi/types/mod.rs

```diff
@@ -0,0 +1,77 @@
+use crate::types;
+use std::{slice, str};
+
+pub mod visitor;
+
+/// # Safety
+///
+/// See the safity documentation of [`slice::from_raw_parts`].
+#[no_mangle]
+pub unsafe extern "C" fn parse_idl(idl_ptr: *const u8, idl_len: u32) -> *mut Program {
```

Comment:
There is no need to make it `pub` and `unsafe` generally if you use it only for FFI. Or do you want to underline that it contains unsafe code?

```suggestion
extern "C" fn parse_idl(idl_ptr: *const u8, idl_len: u32) -> *mut Program {
```

Reply:
Clippy wants it to be marked unsafe as it dereferences raw ptr. `pub` doesn't matter much in this case, but why not